src/cli/abstraction/mod.rs

```diff
@@ -22,6 +22,21 @@ pub struct DatabaseSelectionArguments {
 	pub(crate) database: String,
 }
 
+#[derive(Args, Debug)]
+pub struct DatabaseSelectionOptionalArguments {
```

Comment:
Did you consider reusing `DatabaseSelectionArguments`? It could just be `DatabaseSelectionArguments<T>` since both namespace and database use the same type.